sdk-extensions/jfr-events/src/main/java/io/opentelemetry/sdk/extension/jfr/JfrSpanProcessor.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.jfr;
+
+import static java.util.Objects.nonNull;
+
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Span processor to create new JFR events for the Span as they are started, and commit on end.
+ *
+ * <p>NOTE: JfrSpanProcessor must be running synchronously to ensure that duration is correctly
```

Comment:
We have the https://github.com/open-telemetry/opentelemetry-java/tree/master/sdk-extensions/async-processor which could batch things etc. so that the timing for the of JFR event which is handled in the processor would diverge from the timing of the Span itself.

Reply:
I see - I forgot about that and think many readers would get more confusion than clarity. If we keep a message, making it more generic might make sense - "JfrSpanProcessor measures the timing of events, so it doesn't make sense to wrap it with any other `SpanProcessor` which may affect timings. When possible, register it first before any other processors to allow best measurements." Maybe something like that?